### PR TITLE
Issue11 new structure

### LINKED ISSUE #11: Change structure of nextsim docker images
# New structure (linear)

#### Production
* from ubuntu:bionic
* minimal apt libs
* boost (use builder image with gcc, etc)
* pets
* gmsh
* minimal packaged conda  environment (use continium/miniconda to build and pack, move here some packages from docker-boost-petsc-gmsh-conda)

#### Development
* from continuum:miniconda
* all apt libs (move all apt libs from nextsim here)
* boost
* pets
* gmsh
* full conda environment (move here from docker-boost-petsc-gmsh-conda)

### nextsim
* from docker-nextsim-base (either dev or pro)
* bamg
* mapx
* nextsim (installed in image)

### nextsim-tools
* from nextsim  (either dev or pro)
* mapx
* bamg
* pynextsim (installed in image)

### nextsimf
* from nextsim-tools (either dev or pro)
* pynextsimf

### nextsim-env
* from nextsimf (either dev or pro)
* configuration
* download scripts

# Actions:

## docker-nextsim-base

* rename docker-nextsim-base

### Production
* Move as few apt libs from nextsim
* Update Dockerfile to run in one go (replace `FROM boos_pets_gmsh:base with `FROM base`
* Add minimal conda env using packaging

### Development
* Move all apt libs from nextsim
* Update Dockerfile to run in one go (replace `FROM boos_pets_gmsh:base with `FROM base`
* Add all libs from docker-boost-petsc-gmsh-conda

## nextsim
* install nextsim inside image
* make sure locally compiled nextsim can also run

## nextsim-tools
* install pynextsim inside image

## nextsimf
* install pynextsimf inside image

## nextsim-env
* Add Dockerfile

### PR DESCRIPTION
**Pull request by akorosov**
Closes #11 
* Only one Dockerfile
* Two images: pro and dev

### CODE REVIEW

**Comment by akorosov** [comment]
@tdcwilliams , can you have a look? I tested with nextsim and it can be compiled both from dev and pro images.
I'll push the images to dockerhub soon.

File: environment_pro.yml [hunk 1]
```diff
@@ -1,30 +1,21 @@
...
-  - mock
   - netcdf4
   - netcdftime
-  - nose
-  - notebook
   - numpy
   - pandas
-  - paramiko
   - pillow
   - pygrib
```
Comment (tdcwilliams) [inline environment_pro.yml:15]: not needed
```suggestion
```

File: environment_pro.yml [hunk 2]
```diff
@@ -1,30 +1,21 @@
...
   - pyproj
   - pyshp
   - pyyaml
   - rtree
```
Comment (tdcwilliams) [inline environment_pro.yml:19]: not needed
```suggestion
```

File: environment_pro.yml [hunk 3]
```diff
@@ -1,30 +1,21 @@
...
   - netcdftime
-  - nose
-  - notebook
   - numpy
   - pandas
-  - paramiko
   - pillow
   - pygrib
   - pyproj
   - pyshp
```
Comment (tdcwilliams) [inline environment_pro.yml:17]: only needed if they want to process shapefiles eg from NIC
```suggestion
```

**Review by tdcwilliams — APPROVED** [review]
Looks good - suggested some other packages to be removed; I would recommend changing `pro` to something else (`basic`, or `prod`, since pro = professional usually)

**Pull request merged**